Client/Frontend/Browser/Favicons/FaviconFetcher.swift

```diff
@@ -345,7 +345,9 @@ class FaviconFetcher {
     private func parseHTMLForFavicons(for url: URL, _ completion: @escaping (FaviconAttributes) -> Void) {
         let pageTask = session.dataTask(with: url) { [weak self] (data, response, error) in
             guard let self = self else { return }
-            guard let data = data, error == nil, let root = try? HTMLDocument(data: data) else {
+            guard let data = data, error == nil,
+                  let root = try? HTMLDocument(data: data),
+                  let url = response?.url else {
```

Comment:
are these `url` values needed? can we replace with empty assignment?

Reply:
Yes they are used when creating the resulting `URL` objects as they use `URL(string:relativeTo:)` to ensure we handle possible relative URLs

Reply:
We dont need to worry about this being nil, `response` should be non-nil if `data` is error is `nil`